### PR TITLE
fix(suite): fix fw type switch for old devices

### LINKED ISSUE #5929: BTC only FW offered even when unavailable
**Describe the bug**
BTC only FW offered even when unavailable.
See video and screenshot. Bictoin only firmware is available only since 2019.
We offer it even when does not exist and user is using old firmware

**Steps to reproduce:**
1. Connect device with firmware 2.0.5
2. Navigate to Settings/device
3. Press `Switch to Bitcoin` and proceed with with upgrade.

**Info:**
 - Suite version: desktop 22.8.1 (503b472935dec8a9c07aac5eef5ba43f86780a38)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/22.8.1 Chrome/100.0.4896.160 Electron/18.3.5 Safari/537.36
 - OS: MacIntel
 - Screen: 1680x1050
 - Device: model T 2.0.5  (revision null)
 - Transport: bridge 2.0.31

**Screenshots:**
Screenshot 2022-08-04 at 9 19 29

https://user-images.githubusercontent.com/31506317/182789016-e2d573b4-62bb-4d99-880f-e6b6c3deb023.mov

### PR DESCRIPTION
**Pull request by komret**
- Disable installing Bitcoin-only firmware for T2 devices before 2.0.8.
- Enable installing Bitcoin-only firmware for T1 devices before intermediary firmware.

## Description

Older model T devices (prior to v 2.0.8) need to upgrade to v2.1.1 before they can install the latest version, as explained [here](https://github.com/trezor/trezor-suite/issues/4262). The proposed solution in abec4571e240d155e8eeaa89ad3acc8758d32765 is to offer an upgrade instead of type switch when a user with outdated firmware clicks "Switch to Bitcoin". This is not great UX, but I think it's justifiable for this edge case. An alternative would be to disable or hide this option, but users would still be able to open the modal directly via the URL.

Older model One devices (prior to v1.6.3 or so) need to upgrade to intermediary firmware before automatically upgrading to the latest. Commit 93267f8fa507431b3a84b287416be87657d8a254 correctly sets the target type of the second installation.

## Related Issue

Resolve #5929

## Screenshots (if appropriate):

chrome-capture-2022-7-4

### CODE REVIEW

**Comment by komret** [comment]
@b-irving I added a new string to the modal - this will be in the release:
_Before you can switch to Bitcoin-only, we need to upgrade your firmware to a newer version._

**Comment by b-irving** [comment]
@komret looks good to me

File: packages/suite/src/components/firmware/FirmwareInitial.tsx [hunk 1]
```diff
@@ -114,6 +119,11 @@ export const FirmwareInitial = ({
         return <ConnectDevicePromptManager device={device} />;
     }
 
+    // Bitcoin-only firmware is only available on T2 from v2.0.8 - older devices must first upgrade to 2.1.1 which does not have a Bitcoin-only variant
+    const isBitcoinOnlyAvailable =
+        !!device.firmwareRelease &&
+        !versionUtils.isEqual(device.firmwareRelease?.release.version, [2, 1, 1]);
```
Comment (matejkriz) [inline packages/suite/src/components/firmware/FirmwareInitial.tsx]: 2.1.1 is not the only one without bitcoin only. Anything older than 2.1.5 or 1.8.3 miss it.
 We could just look if `url_bitcoinonly` is available so it's more general and easier to read?
```suggestion
        !!device.firmwareRelease?.release.url_bitcoinonly;
```

Comment (matejkriz) [reply packages/suite/src/components/firmware/FirmwareInitial.tsx]: aha, now I see why it's like that - we need to look if the available update version has url_bitcoinonly and not the current one

Comment (matejkriz) [reply packages/suite/src/components/firmware/FirmwareInitial.tsx]: but that is exactly what you could find in `device.firmwareRelease.release` - info about new the available update version (i.e. info about 2.1.1 if you have 2.0.8), so I believe my suggestion is still valid

Comment (komret) [reply packages/suite/src/components/firmware/FirmwareInitial.tsx]: Good thinking! 14e65bfedb05029f2446291a20a68970ad29c0f9

**Review by matejkriz — APPROVED** [review]
👍 nice

@komret please squash&merge

**Pull request merged**